### PR TITLE
Update/material ui pickers

### PR DESCRIPTION
**Pull request by sjaymoon15**
Implementation of Issue #123

### CODE REVIEW

File: client/src/components/Filter.js [hunk 1]
```diff
@@ -1,306 +1,293 @@
...
+    setDateStart(formattedDate);
+  };
 
-	function handleDateEndChange(e){
-		setDateEnd(e.target.value || dateEndDefault)
-	}
+  const handleDateEndChange = date => {
+    const formattedDate = dateformat(date, 'yyyy-mm-dd');
+    setDateEnd(formattedDate);
+  };
```
Comment (sjaymoon15) [inline client/src/components/Filter.js]: Changed `handleDateStartChange` and `handleDateEndChange` for the date picker change.

File: client/src/components/Filter.js [hunk 2]
```diff
@@ -1,306 +1,293 @@
...
+            format='MM/dd/yyyy'
+            margin='normal'
+            id='end-date-picker'
+            label='End Date'
+            value={dateEnd}
+            onChange={handleDateEndChange}
+            KeyboardButtonProps={{
+              'aria-label': 'change date'
+            }}
+            className={classes.dateInput}
```
Comment (sjaymoon15) [inline client/src/components/Filter.js:280]: Replaced the current date picker with material ui date picker. `OutlinedInput`s were replaced with `KeyboardDatePicker`

File: client/src/components/Filter.js [hunk 3]
```diff
@@ -1,306 +1,293 @@
...
+import MenuItem from '@material-ui/core/MenuItem';
+import FilterModel from '../models/Filter';
+import dateformat from 'dateformat';
+import GSInputLabel from './common/InputLabel';
+import classNames from 'classnames';
+import DateFnsUtils from '@date-io/date-fns';
+import {
+  MuiPickersUtilsProvider,
+  KeyboardDatePicker
+} from '@material-ui/pickers';
```
Comment (sjaymoon15) [inline client/src/components/Filter.js:19]: Libraries added for this update.

**Comment by sjaymoon15** [comment]
Thanks for testing and checking for different behaviors. Will look into this soon !

**Comment by sjaymoon15** [comment]
1. because the new component can not set year directly( I'm correct?), so we must be sure we can input the date string in the input field, now, it does not work, I get error when I delete chars
=> You are correct.
=> And This has been added in the last commit

2.you might notice that I set the date (1970-now) by default, actually this is bad implement, I have to do so because the old component is ugly, I must set some date to avoid some kind of error( I forgot the detail), so if you can remove the default date ( that means if users do not set date proactively, then date is not a filter condition) , that would be wonderful!
=> This has been added in the last commit

3. when I click a date on the component, why it can not select the date and disappear? yes, the date was selected, but the component can't disappear after clicking.
I'm using Chrome 79
=> In order to make the calendar disappear after clicking a date, you need to click anywhere outside of the calendar (or simply press ESC key). This seems to be the default behavior of the date picker library. (https://material-ui.com/components/pickers/#material-ui-pickers)


Would love to get some feedback on these changes.
Especially for number 2 change, in the Filter.js file (client/src/models/Filter.js),rn It does not seem to handle the case when both start date and end date are falsy.

**Comment by dadiorchen** [comment]
I will check the number 2 on my pc, I thought you are correct, Filter.js maybe do not support this change

about the UX of this component, if found a demo on materialUI
Untitled_ Jan 18 2020 9_07 AM
, maybe we can follow this one?  If we use this demo, I thought it will benefit both number 1, and number 3, support year selection and click 'OK' to finish the operation.\

thanks for your efficient work!

**Comment by sjaymoon15** [comment]
Update has been pushed to this PR. 
I replaced the Date picker inline with the Date picker dialog which is the one you shared in the last comment. I like that option to also pick the year as well.  @dadiorchen

**Comment by dadiorchen** [comment]
thanks! I think it's good enough to merge to reop, although there might be tiny bugs in this function, let's solve it in future when we get some bugs, ( mostly it's because we changed the logical, remove the default date as a filter condition)

**Comment by dadiorchen** [comment]
solved #123

**Pull request merged**